### PR TITLE
[HDInsight] Added additional parameters for security profile

### PR DESCRIPTION
**Pull request by wawon-msft**
Note: this API change has not yet been deployed RP-side. I am opening this pull request beforehand to get this API change approved.

We are adding optional parameters to the cluster resource, but we do not expect this to be a breaking change, as this is not modifiable after the initial PUT.

-----

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by marstr** [comment]
@RikkiGibson, have any insights into the Java errors seen above?

**Comment by RikkiGibson** [comment]
Nothing on the top of my head, wonder if @jianghaolu has any idea

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json [hunk 1]
```diff
@@ -446,6 +446,14 @@
                         "type": "string"
                     },
                     "description": "Optional. The Distinguished Names for cluster user groups"
+                },
+                "aaddsResourceId": {
```
Comment (KrisBash) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json]: Keep in mind – there is some risk in adding properties without incrementing the API version. This largely  depends on the service behavior. In cases of update, a down-level client may PUT/PATCH the resource omitting these added properties (which may have been set by a newer client/portal) while updating other properties. In this case, what happens to the resource? Will the behavior be understood by the user? Incrementing API version enables you to have more control in these cases – such as blocking updates by older API versions.

**Review by KrisBash — APPROVED** [review]
Please reference comments about adding properties within an API version

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json (hunk 1 above)
Comment (wawon-msft) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json]: This should be safe, as we don't allow PUTs or PATCHs (other than tags) on this resource after creation, and will definitely not enable such modifications until the next API version.

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
I'm going to move ahead, even though the Java integration is still failing. Are you ready for me to merge, @wawon-msft?

**Comment by marstr** [comment]
Closing this PR as stale. Feel free to have it re-opened or to create a new PR and reference this one for review purposes.

**Pull request closed without merging**